### DISCUSSION TITLE
Adding new Packages to Pyodide Standard distribution

### DISCUSSION THREAD

**Discussion by dmondev** [pyodide/pyodide#1397] [General]
What is the policy for incorporating new packages into the standard pyodide distribution ? Popularity, stability ?

I'm starting this discussion as I recently proposed adding plotly to the standard packages, and I see requests for other packages in the issues as well.

**Comment by hoodmane** [comment]
@rth We should add an answer to this to the docs, perhaps here:
https://pyodide.org/en/latest/development/new-packages.html
and in the faq.

**Comment by rth** [comment]
Historically we have accepted all packages if someone went through the trouble of doing the packaging work. Some package requests were closed due an estimated large maintenance work https://github.com/pyodide/pyodide/issues/50#issuecomment-626007707 but that's debatable.

**For packages with C extensions** I would say it's still a cost / benefit analysis, as in,
 - how useful it would be to users
 - vs how much build time it adds to the CI for everyone and how much effort are we expected to spend to maintain that build. 

https://github.com/pyodide/pyodide/issues/1295 might mitigate some of that, and generally this touches the question of how can we scale to more packages. https://github.com/pyodide/pyodide/issues/795 might be part of the solution, in addition to possibly not building all the packages for each PR as we are doing now (would require to generalize https://github.com/pyodide/pyodide/pull/1262). So there are still some open question on this end.

**For pure Python packages** I would say those should preferably be installed with micropip exclusively https://github.com/pyodide/pyodide/issues/643, unless they require heavy patching or are a dependencies of existing packages in the pyodide distribution.

Here the alternative is between,
 1. add pure Python packages to pyodide and spend time updating them. There is the  `bin/pyodide mkpkg --update` command to automate it somewhat, but it still requires work to maintain.
 2. Improve micropip to address most common issues with this use case

I tend to think that 2 is a better use of our time. We will never be able to add a reasonable part of 235k packages on PyPi so we need an efficient way of installing them from PyPi in any case.

**Reply by grimmerk** [reply]
@rth 
`For pure Python packages`
Compare`pyodide package` vs `python normal package loaded from micropop`, which one has better performance and which one has a smaller downloading size?

**Reply by rth** [reply]
I would expect both aspects to be roughly equivalent but benchmarks are needed.

**Reply by grimmerk** [reply]
@rth  I thought my opend issue, https://github.com/pyodide/pyodide/issues/1426 might be better in `Discussions` but that issue can not be deleted. How do you think?

**Reply by rdb** [reply]
I don't know?  I'm not involved with the Pyodide project.

**Reply by rth** [reply]
@grimmer0125 No worries about it. We can keep it as an issue. Not  being able to easily convert issues to and from  discussion (without keeping duplicate issues/discussion) is somewhat confusing.

**Reply by grimmerk** [reply]
> I don't know? I'm not involved with the Pyodide project.

Sorry, I chose the wrong account.
